### PR TITLE
Add cancancan

### PR DESCRIPTION
**Pull request by andrezafeu**
Add cancancan authorization library. All permissions can be defined in one (or more) ability file keeping all the permissions logic organized.

In the controllers, `load_and_authorize_resource` notifies cancancan that it should authorize users before the actions. The rest I think it's pretty much self explanatory.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -3,4 +3,14 @@ class ApplicationController < ActionController::Base
   # since we are using rails as an API, this may be fine,
   # but we can make clearer decisions when we start creating users etc.
   protect_from_forgery with: :null_session
+
+  rescue_from CanCan::AccessDenied do |exception|
+    respond_to do |format|
+      format.json { head :forbidden, content_type: 'text/html' }
+      format.html { redirect_to main_app.root_url, notice: exception.message }
+      format.js   { head :forbidden, content_type: 'text/html' }
+    end
```
Comment (brycemcd) [inline app/controllers/application_controller.rb]: This is good, but let's limit the responses to just `json` for now please

File: app/models/ability.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  def initialize(user)
+    # Define abilities for the passed in user here.
+
+    # In development, pass :admin => true into user to get admin priviledges
+    user ||= User.new # guest user (not logged in)
+
+    if user.admin?
+      can :manage, :all # user can perform any action on any object
+    else
+      can :manage, BillSummary
```
Comment (brycemcd) [inline app/models/ability.rb]: I don't think we want user's managing Bill Summaries. We do want them to be able to vote though. Given the state of #23, that may come later. `can [:show], Campaign` is probably sufficient for now.

**Comment by andrezafeu** [comment]
@brycemcd I updated the code to reflect the changes suggested by you

**Comment by brycemcd** [comment]
:ship: :it:

**Comment by e91096** [comment]
marking this `waiting for dev` bc of rebase conflicts! otherwise it looks like @brycemcd approved it, and i'm happy to take a final look once conflicts are resolved!

**Comment by andrezafeu** [comment]
Hm... I asked him about it before and he didn't mention anything about merging.. @brycemcd just to confirm, can I rebase and merge this? Or is there any reason why this has been standing by?

**Comment by brycemcd** [comment]
rebase and merge away! IIRC, it was standing by simply because perms are not the highest priority thing right now && we had some work in the auth/auth branch that this depended on. I see no reason to let this sit now.

**Pull request merged**